services/clients/anyscale_client.py

```diff
@@ -76,6 +77,20 @@ async def generate_json(
         )
 
         return llm_json_response
+    
+    async def generate_embedding(
```

Comment:
**curious:** Does OpenAI also provides embeddings with similar inputs/outputs?

If so, I can merge these into the LlmClient interface later so we can swap between OpenAI and Anyscale.

Reply:
Both anyscale embedding models output vectors with 1028 dimensions and the vector database is constrained to vectors with 1028 dimensions. You can't really compare vectors of different dimensions.

OpenAI embedding models state the following

https://platform.openai.com/docs/guides/embeddings/what-are-embeddings
> By default, the length of the embedding vector will be 1536 for text-embedding-3-small or 3072 for text-embedding-3-large. You can reduce the dimensions of the embedding by passing in the [dimensions parameter](https://platform.openai.com/docs/api-reference/embeddings/create#embeddings-create-dimensions) without the embedding losing its concept-representing properties. We go into more detail on embedding dimensions in the [embedding use case section](https://platform.openai.com/docs/guides/embeddings/use-cases).

We can look deeper into constraining the OpenAI embeding models to 1028 dimensions and test the quality differences.